python/ray/serve/_private/http_proxy.py

```diff
@@ -720,8 +739,7 @@ def __init__(
             component_name="http_proxy", component_id=node_ip_address
         )
 
-        if http_middlewares is None:
-            http_middlewares = []
+        http_middlewares = [Middleware(RequestIdMiddleware)]
```

Comment:
I think this needs to extend `http_middlewares` if passed in (though that should get deprecated at some point)